src/effects/effectchain.cpp

```diff
@@ -1,374 +1,430 @@
-    return m_enabledInputChannels.contains(handleGroup);
+    connect(pEnableControl.get(),
+            &ControlObject::valueChanged,
+            this,
+            [this, handleGroup](double value) { slotChannelStatusChanged(value, handleGroup); });
 }
 
-void EffectChain::disableForInputChannel(const ChannelHandleAndGroup& handleGroup) {
-    if (m_enabledInputChannels.remove(handleGroup)) {
-        if (!m_bAddedToEngine) {
-            return;
-        }
-        EffectsRequest* request = new EffectsRequest();
-        request->type = EffectsRequest::DISABLE_EFFECT_CHAIN_FOR_INPUT_CHANNEL;
-        request->pTargetChain = m_pEngineEffectChain;
-        request->DisableInputChannelForChain.pChannelHandle = &handleGroup.handle();
-        m_pEffectsManager->writeRequest(request);
-
-        emit channelStatusChanged(handleGroup.name(), false);
+EffectSlotPointer EffectChain::getEffectSlot(unsigned int slotNumber) {
+    //qDebug() << debugString() << "getEffectSlot" << slotNumber;
+    VERIFY_OR_DEBUG_ASSERT(slotNumber <= static_cast<unsigned int>(m_effectSlots.size())) {
+        return EffectSlotPointer();
     }
+    return m_effectSlots[slotNumber];
 }
 
-double EffectChain::mix() const {
-    return m_dMix;
-}
-
-void EffectChain::setMix(const double& dMix) {
-    m_dMix = dMix;
-    sendParameterUpdate();
-    emit mixChanged(dMix);
+void EffectChain::slotControlClear(double v) {
+    for (EffectSlotPointer pEffectSlot : std::as_const(m_effectSlots)) {
+        pEffectSlot->slotClear(v);
+    }
 }
 
-EffectChainMixMode EffectChain::mixMode() const {
-    return m_mixMode;
-}
+void EffectChain::slotControlChainSuperParameter(double v, bool force) {
+    // qDebug() << debugString() << "slotControlChainSuperParameter" << v;
 
-void EffectChain::setMixMode(EffectChainMixMode mixMode) {
-    m_mixMode = mixMode;
-    sendParameterUpdate();
-    emit mixModeChanged(mixMode);
+    m_pControlChainSuperParameter->set(v);
+    for (const auto& pEffectSlot : std::as_const(m_effectSlots)) {
+        pEffectSlot->setMetaParameter(v, force);
+    }
 }
 
-void EffectChain::addEffect(EffectPointer pEffect) {
-    //qDebug() << debugString() << "addEffect" << pEffect;
-    if (!pEffect) {
-        // Insert empty effects to preserve chain order
-        // when loading chains with empty effects
-        m_effects.append(pEffect);
-        return;
+void EffectChain::slotControlChainSelector(double value) {
+    int index = presetIndex();
+    if (value > 0) {
+        index++;
+    } else {
+        index--;
     }
+    loadChainPreset(presetAtIndex(index));
+}
 
-    VERIFY_OR_DEBUG_ASSERT(!m_effects.contains(pEffect)) {
+void EffectChain::slotControlLoadedChainPresetRequest(double value) {
+    // subtract 1 to make the ControlObject 1-indexed like other ControlObjects
+    int index = static_cast<int>(value) - 1;
+    if (index < 0 || index >= m_pChainPresetManager->numPresets()) {
```

Comment:
```suggestion
    if (index < 0 || index >= numPresets()) {
```
accessing `m_pChainPresetManager` directly, bypasses the `QuickEffectChain::numPresets` override. This results in only the EffectChainPresets being available to load via the CO (while in case of the quickeffect, we also want regular effects to be loaded).